### PR TITLE
added listing of discarded products

### LINKED ISSUE #105: listing of discarded products

### PR DESCRIPTION
**Pull request by aatma001**
resolves #105

### CODE REVIEW

File: src/components/WelcomePage.tsx [hunk 1]
```diff
@@ -202,6 +208,15 @@ class WelcomePage extends React.Component<Props, any> {
                   />
                 )}
               />
+              <Route
+                path="/discards"
+                exact={ true }
```
Comment (lauriLukkarinen) [inline src/components/WelcomePage.tsx:213]:
```suggestion
                exact
```

File: src/components/WelcomePage.tsx [hunk 2]
```diff
@@ -250,13 +265,30 @@ class WelcomePage extends React.Component<Props, any> {
...
+                    keycloak={ this.state.keycloak }
+                    discardId={ props.match.params.discardId as string }
+                  />
+                )}
+              />
               <Route
                 path="/createPacking"
                 exact={true}
                 render={props => (
                   <CreatePacking keycloak={this.state.keycloak}/>
```
Comment (lauriLukkarinen) [inline src/components/WelcomePage.tsx:282]:
```suggestion
                render={ props => (
                  <CreatePacking keycloak={ this.state.keycloak }/>
```

File: src/localization/fi.json [hunk 3]
```diff
@@ -221,5 +221,11 @@
     "packingTableHeaderPackageSize": "Pakkauskoko",
     "allPackingStates": "Kaikki tilat",
     "allProductionLines": "Kaikki linjat",
-    "totalPackingsRow": "Yhteensä: {0} laatikkoa"
-  }
\ No newline at end of file
+    "totalPackingsRow": "Yhteensä: {0} laatikkoa",
+
+    "discards": "Poisheitetyt tuotteet",
```
Comment (lauriLukkarinen) [inline src/localization/fi.json:226]:
```suggestion
    "discards": "Poisheitto kylmiöstä",
```

File: src/localization/en.json [hunk 4]
```diff
@@ -222,5 +222,12 @@
   "packingTableHeaderPackageSize": "Package size",
   "allPackingStates": "All states",
   "allProductionLines": "All lines",
-  "totalPackingsRow": "Total: {0} boxes"
+  "totalPackingsRow": "Total: {0} boxes",
+
+  "discards": "Discarded products",
```
Comment (lauriLukkarinen) [inline src/localization/en.json:227]:
```suggestion
  "discards": "Discarded from storage",
```

File: src/types/index.tsx [hunk 5]
```diff
@@ -41,5 +42,15 @@ export interface StoreState {
 export interface ErrorMessage {
   title?: string,
   message?: string,
-  exception?: Error
+  exception?: 
+  Error
+}
```
Comment (lauriLukkarinen) [inline src/types/index.tsx:47]:
```suggestion
  exception?: Error
}
```

**Review by lauriLukkarinen — CHANGES_REQUESTED** [review]

File: src/components/DiscardList.tsx [hunk 6]
```diff
@@ -0,0 +1,410 @@
...
+    }
+  }
+   
+  /**
+   * Render 
+   */
+  public render() {
+    const { filters, discardedProducts, products, loading } = this.state;
+
+    const possibleLoader = (): any=> {
```
[the remark this answers is not in the document]
Comment (aatma001) [reply src/components/DiscardList.tsx:100]: I want it to be rendered bottom of table

**Pull request merged**